cdk/src/open-data-platform/data-plane/data-import/write-violations-data.handler.ts

```diff
@@ -37,99 +43,144 @@ const COMPLIANCE_STATUS_MAP = new Map([
                 .pwsId(properties[PWSID])
                 .violationCode(properties[VIOLATION_CODE])
                 .complianceStatus(
                   COMPLIANCE_STATUS_MAP.get(properties[COMPLIANCE_STATUS_CODE]) ?? '',
                 )
-                .startDate(startDate.format('YYYY-MM-DD'))
+                .startDate(startDate.format(POSTGRESQL_DATE_FORMAT))
+                .endDate(endDate.format(POSTGRESQL_DATE_FORMAT))
                 .build();
-              results.push(violationsRow);
-
-              // Every row size, write into the db.
-              if (results.length == batchSize) {
-                // Pause reads while inserting into db.
-                fileStream.pause();
-                await insertRows(db, results);
-                results = [];
-                fileStream.resume();
-              }
+
+              tableRows.push(tableRowToInsert);
             }
           }
-        } else if (numberRowsParsed > endIndex) {
-          // Stop reading stream if numberOfRowsToWrite has been met.
+        }
+
+        numberRowsParsed += rows.length;
+        console.log(`Parsed ${numberRowsParsed}`);
+        executeBatchOfRows(db, tableRows, promises);
+
+        // Stop reading stream if numberOfRowsToWrite has been met.
+        if (numberRowsParsed > endIndex) {
           pipeline.destroy();
         }
-        numberRowsParsed += row.length;
       })
       .on('error', (error: Error) => {
         reject(error);
       })
       // Gets called by pipeline.destroy()
       .on('close', async (_: Error) => {
+        await handleEndOfFilestream(promises);
         resolve(numberRowsParsed);
       })
       .on('end', async () => {
+        await handleEndOfFilestream(promises);
         resolve(numberRowsParsed);
       });
   });
 }
 
 /**
- * Parses S3 'violations_by_water_system.csv' file and writes rows
+ * Writes the table rows to the db and adds execution to list of promises.
+ *
+ * @param db: Database to write to.
+ * @param tableRows: Rows to write to the db.
+ * @param promises: List of promises collecting db executions.
+ */
+function executeBatchOfRows(
+  db: PoolClient,
+  tableRows: ViolationsTableRow[],
+  promises: Promise<QueryArrayResult>[],
+) {
+  if (tableRows.length > 0) {
+    const promise = insertRows(db, tableRows);
+    promise.catch((_) => {}); // Suppress unhandled rejection.
+    promises.push(promise);
+  }
+}
+
+/**
+ * Logs number and details of errors that occurred for all inserts.
+ * @param promises of SQL executions.
+ */
+async function handleEndOfFilestream(promises: Promise<QueryArrayResult>[]) {
+  const result = await Promise.allSettled(promises);
+  const errors = result.filter((p) => p.status == 'rejected') as PromiseRejectedResult[];
+  console.log(`Number errors during insert: ${errors.length}`);
+
+  // Log all the rejected promises to diagnose issues.
+  errors.forEach((error) => console.log(error.reason));
```

Comment:
Can you add some string to the log so it's easy to search for? e.g. `console.log('Insert error:', error.reason)`